### PR TITLE
[Browse] Add skeletons to Browse models

### LINKED ISSUE #44111: Add skeletons to Browse models

### PR DESCRIPTION
**Pull request by rafpaf**
This PR
* adds a shimmer effect to the skeletons
* adds a `natural` prop that makes the skeletons have a random width between 50% and 80%
* adds a skeleton to Browse models

image

Closes #44111

### CODE REVIEW

**Comment by rafpaf** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/metabase/metabase/44008?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#43979** <a href="https://app.graphite.dev/github/pr/metabase/metabase/43979?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#44008** <a href="https://app.graphite.dev/github/pr/metabase/metabase/44008?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#43970** <a href="https://app.graphite.dev/github/pr/metabase/metabase/43970?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rafpaf and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: frontend/src/metabase/ui/components/feedback/Skeleton/Skeleton.tsx [hunk 1]
```diff
@@ -1,3 +1,19 @@
...
+
+export const Skeleton = ({
+  natural,
+  ...props
+}: SkeletonProps & {
+  /** Automatically assign a natural-looking, random width to the skeleton */
+  natural?: boolean;
+}) => {
+  const width = useMemo(
+    () => (natural ? `${Math.random() * 30 + 50}%` : props.width),
```
Comment (sloansparger) [inline frontend/src/metabase/ui/components/feedback/Skeleton/Skeleton.tsx:13]:
```suggestion
    () => {
      if (!natural) return props.width;
      // validate naturalMin < max somewhere above 
      const spread = naturalMax - naturalMin;
      return `${_.random(0, spread) + naturalMin}%`
    }
```
Would be cool if the min/max were configurable. Also using lodash to get round numbers might nice.

File: frontend/src/metabase/ui/components/feedback/Skeleton/Skeleton.tsx [hunk 2]
```diff
@@ -1,3 +1,19 @@
-export type { SkeletonProps } from "@mantine/core";
-export { Skeleton } from "@mantine/core";
-export { getSkeletonOverrides } from "./Skeleton.styled";
+import type { SkeletonProps } from "@mantine/core";
+import { Skeleton as MantineSkeleton } from "@mantine/core";
+import { useMemo } from "react";
+
+export const Skeleton = ({
```
Comment (sloansparger) [inline frontend/src/metabase/ui/components/feedback/Skeleton/Skeleton.tsx:5]: (i don't know the answer to this question)
do we need to add storybook stories for new ui components?

File: frontend/src/metabase/ui/components/feedback/Skeleton/Repeat.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  children,
+}: PropsWithChildren<{
+  times: number;
+}>) => {
+  if (!isValidElement(children)) {
+    return null;
+  }
+  return Array.from({ length: times }).map((_, index) => {
+    const props = { key: `${index}` };
+    if (isValidElement(children)) {
```
Comment (sloansparger) [inline frontend/src/metabase/ui/components/feedback/Skeleton/Repeat.tsx:16]: you've already done this check on line 11

File: frontend/src/metabase/browse/components/ModelsTable.styled.tsx [hunk 4]
```diff
@@ -8,11 +8,17 @@ import {
...
-  cursor: pointer;
+export const ModelTableRow = styled.tr<{ skeleton?: boolean }>`
   :outline {
     outline: 2px solid var(--mb-color-brand);
   }
+  ${props =>
+    props.skeleton
+      ? `
+    :hover { background-color: unset ! important; }
+    td { cursor: unset ! important; }
```
Comment (sloansparger) [inline frontend/src/metabase/browse/components/ModelsTable.styled.tsx:19]: i didn't think there could be a space between `!` and `important`. scared me fears some old browser might not like this so i might keep to the spec here.

**Review by sloansparger — APPROVED** [review]
LGTM on the whole, two small things that I think are worth changing before merging and one suggestion

File: frontend/src/metabase/ui/components/feedback/Skeleton/Skeleton.tsx (hunk 2 above)
Comment (kdoh) [reply frontend/src/metabase/ui/components/feedback/Skeleton/Skeleton.tsx:5]: Yeah the protocol is generally to add an entry to Storybook if you're adding a new export from `metabase/ui` to help with component discovery and reuse.

**Pull request merged**